lib/src/rx_ext/serial_disposable.dart

```diff
@@ -9,25 +9,24 @@ class SerialDisposable<T> {
     if (_isDisposed) {
       throw _SerialAlreadyDisposed(runtimeType);
     }
-    if (_value != null) await _dispose(_value);
+    if (_value != null) await _dispose(_value!);
     _value = newValue;
-    return newValue;
   }
 
   /// Dispose underlying resource
   Future<void> dispose() async {
     _isDisposed = true;
     if (_value != null) {
-      await _dispose(_value);
+      await _dispose(_value!);
```

Comment:
It's a pity Dart doesn't "remember" a certain value is not null after an explicit check 😞

Reply:
I think it has only problem with Generic types. Because in other cases this works properly